container/sagemaker/tensorflow-serving.js

```diff
@@ -14,11 +14,30 @@ function invocations(r) {
 }
 
 function ping(r) {
-    // TODO replace with call to Model Status API when Tensorflow Serving 1.12 is available
+    if ('1.11' == r.variables.tfs_version) {
+        return ping_tfs_1_11(r)
+    }
```

Comment:
What is the benefit to keep the 1.11 container logic here?

Reply:
Can you request the version using TFS requests? I am unsure if it comes in the metadata or not.

Reply:
TFS 1.11 does not support the ModelStatus api, so there is a different code path. I want to retain ability to ship updates for both 1.11 and 1.12 -- this seems particularly important for TF since our training users will be split across several versions for a while.

Reply:
Tagging the version achieves the objective mentioned above.

Reply:
no it doesn't -- tagging/branching means we need to maintain multiple branches with code that will diverge over time.

Reply:
I agree with this. Release branching is great, but it comes at an overhead. I'm not sure the overhead is worth it for such a small difference. When there are substantial changes (e.g. see sagemaker-tensorflow-extensions) between versions, then go ahead with release branching.